src/plugins/teleport-plugin-react-styled-components/index.ts

```diff
@@ -0,0 +1,89 @@
+import { ComponentPluginFactory, ComponentPlugin } from '../../typings/generators'
+import { generateStyledComponent } from './utils'
+import { traverseElements, transformDynamicStyles } from '../../shared/utils/uidl-utils'
+import { stringToUpperCamelCase } from '../../shared/utils/string-utils'
+import {
+  createJSXSpreadAttribute,
+  addDynamicAttributeOnTag,
+} from '../../shared/utils/ast-jsx-utils'
+
+interface StyledComponentsConfig {
+  componentChunkName: string
+  importChunkName?: string
+}
+
+export const createPlugin: ComponentPluginFactory<StyledComponentsConfig> = (config) => {
+  const { componentChunkName = 'react-component', importChunkName = 'import-local' } = config || {}
+
+  const reactStyledComponentsPlugin: ComponentPlugin = async (structure) => {
+    const { uidl, chunks, dependencies } = structure
+    const { node } = uidl
+    const componentChunk = chunks.find((chunk) => chunk.name === componentChunkName)
+    if (!componentChunk) {
+      return structure
+    }
+
+    const jsxNodesLookup = componentChunk.meta.nodesLookup
+    const jssStyleMap = {}
+
+    traverseElements(node, (element) => {
+      const { style, key, elementType } = element
+      if (style) {
+        const root = jsxNodesLookup[key]
+        const className = `${stringToUpperCamelCase(key)}`
+        let timesReferred: number = 0
+
+        Object.keys(style).map((item) => {
+          // @ts-ignore-next-line
```

Comment:
Please check this, it is throwing some ts-lint issue I tried to figure out, but a little confused as we are already referring ``UIDLDynamicReference`` which gives types for ``refereceTypes``